### PR TITLE
Add method to notify pump manager of max temp basal changes

### PR DESCRIPTION
**Pull request by ps2**
Pod pump manager needs maximumTempBasal to be able to display relative basal rate visually.
https://tidepool.atlassian.net/browse/LOOP-1154

https://github.com/tidepool-org/Loop/pull/80
https://github.com/tidepool-org/DashKit/pull/26

### CODE REVIEW

File: LoopKit/DeviceManager/PumpManager.swift [hunk 1]
```diff
@@ -163,6 +163,12 @@ public protocol PumpManager: DeviceManager {
     ///   - completion: A closure called after the command is complete
     ///   - error: An error describing why the command failed
     func resumeDelivery(completion: @escaping (_ error: Error?) -> Void)
+    
+    /// Notifies the PumpManager of a change in the user's preference for maximum basal rate.
+    ///
+    /// - Parameters:
+    ///   - rate: The maximum rate the pumpmanager should expect to receive in an enactTempBasal command.
+    func setMaximumTempBasalRate(_ rate: Double)
```
Comment (rickpasetto) [inline LoopKit/DeviceManager/PumpManager.swift:171]: Do you also need to provide a default implementation, so as not to break existing `PumpManager`s?

**Review by rickpasetto — COMMENTED** [review]
Quick question before approval...

File: LoopKit/DeviceManager/PumpManager.swift (hunk 1 above)
Comment (ps2) [reply LoopKit/DeviceManager/PumpManager.swift:171]: Method dispatch for protocol extensions use static dispatch, which means that any clients with a generic PumpManager reference will call the default implementation, and not the overridden one.

Comment (rickpasetto) [reply LoopKit/DeviceManager/PumpManager.swift:171]: I need to understand better what the best practices are, then, for introducing breaking API changes such as this with respect to how it affects downstream dependents.  Probably a topic for a dev sync.

**Review by rickpasetto — APPROVED** [review]
LGTM though I have a question about best practices w.r.t. breaking API changes.

File: LoopKit/DeviceManager/PumpManager.swift (hunk 1 above)
Comment (ps2) [reply LoopKit/DeviceManager/PumpManager.swift:171]: Semantic versioning of LoopKit would require a major version bump on release. The versioning of LoopKit is a bit complicated by the existence of separate DIY and Tidepool forks. During its time on dev, though, we just need to coordinate these changes. This particular change only affects DashKit on the Tidepool side, and I am submitting those changes in tandem with these.

**Pull request merged**